plugins/scaffolder-backend-module-gitlab/report.api.md

```diff
@@ -76,11 +76,17 @@ export const createGitlabProjectDeployTokenAction: (options: {
 }) => TemplateAction<
   {
     name: string;
+    scopes: (
+      | 'read_repository'
+      | 'read_registry'
+      | 'write_registry'
+      | 'read_package_registry'
+      | 'write_package_registry'
+    )[];
```

Comment:
Hm. Is this a good thing being the specific strings? Is this truly the only possible scopes now and forever? :)

Reply:
Up to you guys :) Having it as string allows "unwanted" behaviour for now, but of course the user will quickly realize.

I'm happy to leave it as a `string`. Should I cast it `as DeployTokenScope` in `createGitlabProjectDeployTokenAction` when calling `api.DeployTokens.create`?

Reply:
might be good to list out the known good ones, and have `| string` tagged onto the end. Would allow intellisense to suggest these values, while also providing an escape hatch (and type hints) that there could be more options in the future.

Reply:
Yep that's a good middle ground. Will still need a cast internally but is useful to the end user.

Reply:
Can you give me a hand on this, guys? TS considers some types as "subtypes" of others. For instance:
```typescript
let a: string | "read_repository" | "read_registry" | "write_registry" | "read_package_registry" | "write_package_registry";
```
is equivalent (in VSCode suggestions) to:
```typescript
let a: string;
```

With `zod`, these types would be generated with:
```typescript
scopes: z.array(
  z.union([
    z.enum([
      'read_repository',
      'read_registry',
      'write_registry',
      'read_package_registry',
      'write_package_registry',
    ]),
    z.string()
  ]),
  { description: 'Scopes' },
```

But I get no benefit with intellisense, and the API report will declare `string[]`.

I'm good with reverting it to `string[]`. It looks like the only viable alternative.

Reply:
I'm pushing it with `string[]` for now. If you like it, we continue like that. Otherwise let me know and I change it to `DeployTokenScope[]` (or maybe a 3rd option?)

Reply:
leaving it as `string[]` is probably our best bet